biorxiv/journal_tracker/journal_recommendation_experiment.py

```diff
@@ -0,0 +1,354 @@
+        knn_evaluate, cv=10, 
+        random_state=100
+    )
+    
+    print()
+    
+    result_dict[dim] = fold_predictions
+
+
+# ## KNearestNeighbors Centroid analysis
+
+# Following up on the original idea, I thought a helpful experiment would be to perform a centroid analysis (i.e. take the average of all papers within each journal). Similar to above I trained a KNearestNeighbors classifier to see if the correct journal will appear in the top 10 neighbors.
+
+# In[16]:
+
+
+_ = cross_validation(
+    knn_model, subsampled_training_dataset[300], 
+    knn_centroid_evaluate, cv=10, 
+    random_state=100
+)
+
+
+# ## KNearestNeighbors Centroid Analysis - Full dataset
+
+# This section I'm using the entire dataset to calculate journal centroids and then evaluate performance on the sub-sampled dataset.
+
+# In[19]:
+
+
+predictions, true_labels = (
+    knn_centroid_evaluate(
+        knn_model,
+        full_training_dataset[300], 
+        subsampled_training_dataset[300]
+    )
+)
+
+
+# In[20]:
+
+
+accs = [
+    (
+         1 if true_labels[data_idx] in prediction_row 
+         else 0 
+     )
+     for data_idx, prediction_row in enumerate(predictions)
+]
+
+
+# In[21]:
+
+
+print(f"{np.sum(accs)} out of {len(accs)}")
+print(f"{np.mean(accs)}% correct")
+
+
+# # Golden Set Analysis
+
+# In[22]:
+
+
+biorxiv_embeddings_df = pd.read_csv(
+    Path("../word_vector_experiment/output/word2vec_output/biorxiv_all_articles_300.tsv.xz")
+    .resolve(),
+    sep="\t"
+)
+
+biorxiv_embeddings_df.head()
+
+
+# In[23]:
+
+
+golden_dataset = (
+    golden_set_df[["document", "pmcid"]]
+    .merge(pmc_articles_df[["journal", "pmcid"]], on="pmcid")
+    .merge(biorxiv_embeddings_df, on="document")
+)
+golden_dataset.head()
+
+
+# ## Centroid Analysis
+
+# In[25]:
+
+
+predictions, true_labels = (
+    knn_centroid_evaluate(
+        knn_model,
+        full_training_dataset[300], 
+        golden_dataset.drop(["pmcid", "document"], axis=1)
+    )
+)
+
+
+# In[26]:
+
+
+accs = [
+    (
+         1 if true_labels[data_idx] in prediction_row 
+         else 0 
+     )
+     for data_idx, prediction_row in enumerate(predictions)
+]
+
+
+# In[27]:
+
+
+print(f"{np.sum(accs)} out of {len(accs)}")
+print(f"{np.mean(accs)}% correct")
+
+
+# ## Subsampled Paper Analysis
+
+# In[28]:
+
+
+predictions, true_labels = (
+    knn_evaluate(
+        knn_model,
+        subsampled_training_dataset[300],
+        golden_dataset.drop(["pmcid", "document"], axis=1)
+    )
+)
+
+
+# In[29]:
+
+
+accs = [
+    (
+         1 if true_labels[data_idx] in prediction_row 
+         else 0 
+     )
+     for data_idx, prediction_row in enumerate(predictions)
+]
+
+
+# In[30]:
+
+
+print(f"{np.sum(accs)} out of {len(accs)}")
+print(f"{np.mean(accs)}% correct")
+
+
+# Conclusions for this notebook:
+# 1. Mega-journals cover a wide range of research topics.
+# 2. The correct journal only appears in the top ten about 37-39 percent of the time.
```

Comment:
I'm not sure whether your goal is to max out prediction accuracy or just use a simple model to show that the embeddings are predictive of which journal the articles end up in.

If you're looking to get as accurate as possible, manhattan distance or a fractional distance may outperform euclidean distance [due to high dimensionality](https://bib.dbvis.de/uploadedFiles/155.pdf).

Reply:
>I'm not sure whether your goal is to max out prediction accuracy or just use a simple model to show that the embeddings are predictive of which journal the articles end up in.

The goal is to show that embeddings are predictive of which journal the articles end up; however, I want to be as accurate as possible. Long sentence converted to a short one: the goal is to do both.

>If you're looking to get as accurate as possible, manhattan distance or a fractional distance may outperform euclidean distance due to high dimensionality.

Thanks will see how performance changes.